src/App.jsx

```diff
@@ -14,45 +16,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_LENGHT = 'length';
+const SORT_ALPHABETICALLY = 'alphabetically';
+
+function getPreparedGoods(goods, { sortField, isReversed }) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_LENGHT:
+          return good1.length - good2.length;
+
+        case SORT_ALPHABETICALLY:
+          return good1.localeCompare(good2);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const visibleGoods = getPreparedGoods(goodsFromServer,
+    { sortField, isReversed });
+
+  const handleReset = () => {
+    setIsReversed(false);
+    setSortField('');
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortField(SORT_ALPHABETICALLY)}
+          type="button"
+          className={cn('button is-info',
+            { 'is-light': sortField !== SORT_ALPHABETICALLY })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortField(SORT_LENGHT)}
+          type="button"
+          className={cn('button is-success',
+            { 'is-light': sortField !== SORT_LENGHT })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => setIsReversed(!isReversed)}
+          type="button"
+          className={cn('button is-warning',
+            { 'is-light': isReversed !== true })}
+        >
+          Reverse
+        </button>
+
+        {(sortField === SORT_LENGHT
+        || sortField === SORT_ALPHABETICALLY
+        || isReversed) && (
```

Comment:
To make the code shorter, it's enough to state this {(sortField || isReversed) && ( ...